pkgs/tools/typesetting/tex/texlive/combine.nix

```diff
@@ -298,14 +292,20 @@ in (buildEnv {
 
   # MkIV uses its own lookup mechanism and we need to initialize
   # caches for it.
+  # We use faketime to fix the embedded timestamps and patch the uuids
+  # with some random but constant values.
   ''
     if [[ -e "$out/bin/mtxrun" ]]; then
-      mtxrun --generate
+      substitute "$TEXMFDIST"/scripts/context/lua/mtxrun.lua mtxrun.lua \
+        --replace 'cache_uuid=osuuid()' 'cache_uuid="e2402e51-133d-4c73-a278-006ea4ed734f"' \
+        --replace 'uuid=osuuid(),' 'uuid="242be807-d17e-4792-8e39-aa93326fc871",'
+      FORCE_SOURCE_DATE=1 TZ= faketime -f '@1980-01-01 00:00:00 x0.001' luatex --luaonly mtxrun.lua --generate
```

Comment:
This doesn't matter at all as I assume we just need something constant for reproducibility and the value doesn't matter... but I just saw a build logs fly by and... was there any reason for 19_8_0?  :grin:

(curiosity getting the better of me, I didn't see any comment about it so I thought I'd ask! But a quick grep in the repo has a few others 1980-01-01 so there might actually be a reason...)

Reply:
1980-01-01 is the lowest date representable as 'MS-DOS date', which notably is used in the ZIP file format date fields.

http://justsolve.archiveteam.org/wiki/MS-DOS_date/time